### PR TITLE
Authoring

### PR DESCRIPTION
**Pull request by pjanik**
This PR adds a new page available at `?authoring`. It lets authors customize URL for an activity.
Sidebar panel is now configurable too.

Also, I've started using css-modules, so I guess we could slowly switch to that and maybe refactor other components while working on them. Actualy, sometimes it's the only way to theme react-toolbox components, so that's why I had to use it. But I think it's a nice pattern in general.

### CODE REVIEW

File: css-modules/authoring.less [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    color: #666;
+    top: 24px;
+  }
+
+  .filled ~ .label {
+    color: #555;
+  }
+}
+
+// checkbox theme
```
Comment (ekosmin) [inline css-modules/authoring.less]: I read up on CSS modules - having scoped CSS does seem really nice, I'll try it in my next commit. I guess if the toolbox components are written with css modules, the class names all end up gibberish, so you need to use css modules to style them?

Comment (pjanik) [reply css-modules/authoring.less]: Yes, exactly. Sometimes it's enough to assign className to the top component and then try to use element matches, but quite often it's annoying and I reached the point that I couldn't do that. With CSS modules, you just specify class names that are described in react-toolbox docs and you can theme all the elements.

It makes sense and it won't hurt our code either. No need to care about unique class names.

**Comment by pjanik** [comment]
@ekosmin, good catch, thanks! I was testing with localhost:8080/ but ofc it didn't work with index.html. Fixed now.

I had to force push again and merge master into authoring branch again. Not sure how you merged changes, but it looked strange and you force pushed your changes, so I got in a state where I couldn't push my changes at all. Not sure why, but it wasn't worth more investigation. Nice way to do it is to merge master into topic branch, so they no force pushing is necessary. Github website has pretty nice UI for it actually, so there's even no need to do it locally.

**Comment by ekosmin** [comment]
Sorry about the forced changes, I've never used the Github merge tools before, but I'll give them a try next time! I think this looks good now - I had to change the default value of `modelId` since otherwise there was a NPE from a missing constructor. Do we want to be able to accept null defaults in config? Either way, I'm going to go ahead and merge this.

**Pull request merged**